### PR TITLE
Remove transactions from pool when they're committed to a block

### PR DESCRIPTION
**Pull request by enolan**
Remove transactions from the pool when we see a new best tip that contains them. This is a hacky solution to unblock running longer PoS tests with transactions. There are a few things wrong with it: removing transactions is super inefficient, we don't handle backtracking in external transitions, and we don't do anything special when bootstrapping is engaged. See also #1734.

Tested at the command line by running a proposer and submitting transactions, then watching the log to verify they're dropped. There's a commented out automated test that is broken because of #1748.

Checklist:

- [ ] Tests were added for the new behavior
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

### CODE REVIEW

File: src/lib/network_pool/network_pool.ml [hunk 1]
```diff
@@ -44,13 +54,20 @@ module type Network_pool_intf = sig
...
 
+module type Transition_frontier_intf = sig
+  type t
+end
+
 module Snark_pool_diff = Snark_pool_diff
 
 module Make
-    (Pool : Pool_intf)
+    (Transition_frontier : Transition_frontier_intf)
```
Comment (bkase) [inline src/lib/network_pool/network_pool.ml]: if it's just `sig type t end`, can you inline this?

Comment (bkase) [reply src/lib/network_pool/network_pool.ml]: nit

File: src/lib/transaction_pool/transaction_pool.ml [hunk 2]
```diff
@@ -28,19 +58,109 @@ module Make (Payment : sig
...
+    in
+    don't_wait_for
+      ( fst
+      @@ Broadcast_pipe.Reader.iter frontier_broadcast_pipe
+           ~f:(fun frontier_opt ->
+             match frontier_opt with
+             | None -> (
+                 Logger.debug t.log "no frontier" ;
+                 (* Sanity check: the view pipe should have been closed before
+                    the frontier was destroyed. *)
```
Comment (bkase) [inline src/lib/transaction_pool/transaction_pool.ml]: I don't think this is hooked up properly yet. Look at the TODO on `close` in Transition-frontier.

Comment (enolan) [reply src/lib/transaction_pool/transaction_pool.ml]: It gets closed in transition_router.ml:206. I'll go remove the comment.

**Pull request merged**